### PR TITLE
Fix unmanaged calling convention defaults

### LINKED ISSUE #363: Generating native code doesnt work with Grpc
Good day!
I`m trying to build simple Grpc example with Microsoft.Dotnet.ILCompiler, but it fails with following errors:

> GreeterServer -> D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\bin\Release\netcoreapp3.1\win-x64\GreeterServer.dll

>   Generating compatible native code. To optimize for size or speed, visit https://aka.ms/OptimizeCoreRT
> GreeterServer.obj : error LNK2001: unresolved external symbol grpcsharp_init [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol grpcsharp_shutdown [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol grpcsharp_version_string [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol grpcsharp_batch_context_create [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol grpcsharp_batch_context_recv_initial_metadata [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol grpcsharp_batch_context_recv_message_length [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol grpcsharp_batch_context_recv_message_next_slice_peek [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol grpcsharp_batch_context_recv_status_on_client_status [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol grpcsharp_batch_context_recv_status_on_client_details [D:\GIT

> 
> <....>
> 
> GreeterServer.obj : error LNK2001: unresolved external symbol gprsharp_convert_clock_type [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol gprsharp_sizeof_timespec [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol grpcsharp_test_callback [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol grpcsharp_test_nop [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol grpcsharp_test_override_method [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol grpcsharp_test_call_start_unary_echo [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol dlopen [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol dlerror [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> GreeterServer.obj : error LNK2001: unresolved external symbol dlsym [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> bin\Release\netcoreapp3.1\win-x64\native\GreeterServer.exe : fatal error LNK1120: 100 unresolved externals [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]
> D:\build\Microsoft.NETCore.Native.targets(331,5): error MSB3073: The command ""C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.27.29110\bin\Hostx64\x64\link.exe"  @"obj\Release\netcoreapp3.1\win-x64\native\link.rsp"" exited with code 1120. [D:\GIT\grpc-master\examples\csharp\Helloworld\GreeterServer\GreeterServer.csproj]`

Doesnt work with neither Microsoft.DotNet.ILCompiler 1.0.0-alpha-29408-02 or 6.0.0-alpha.1.20566.4
I`ve attached the project which fails as an example for reproduction. 
[grpc_fail_example.zip](https://github.com/dotnet/runtimelab/files/5559752/grpc_fail_example.zip)

Steps to reproduce: 
1. unzip archive and cd to GreeterServer or GreeterClient
2. run command `dotnet publish -c Release -r win-x64`

Any help will be appreciated.
Thanks in advance !

### PR DESCRIPTION
**Pull request by jkotas**
The default calling unmanaged convention turned into managed calling convention in some places.

Fixes #363

### CODE REVIEW

**Review by MichalStrehovsky — APPROVED** [review]
Thanks! Do you know if we need a regression test or is this covered through one of the interop tests that we currently don't run?

**Comment by jkotas** [comment]
It is probably tested, but it does not hurt to add it to smoke tests.

File: src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs [hunk 1]
```diff
@@ -1055,10 +1052,19 @@ private bool isIntrinsicType(CORINFO_CLASS_STRUCT_* classHnd)
             return type.IsIntrinsic;
         }
 
+        private MethodSignatureFlags PlatformDefaultUnmanagedCallingConvention()
+        {
+            return _compilation.TypeSystemContext.Target.IsWindows ?
+                MethodSignatureFlags.UnmanagedCallingConventionCdecl : MethodSignatureFlags.UnmanagedCallingConventionCdecl;
```
Comment (andrew-boyarshin) [inline src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs:1058]: @jkotas is it intended, that both conditional expressions are `Cdecl`? `IsWindows` is a redundant check then. Perhaps one should be `stdcall`.

Comment (jkotas) [reply src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs:1058]: Good catch! Thank you!

#376

**Pull request merged**